srml/elections-phragmen/src/lib.rs

```diff
@@ -0,0 +1,957 @@
+//! always trigger a new election round.
+//!
+//! Voters can vote for as many of the candidates by providing a list of account ids. Invalid votes
+//! (voting for non-candidates) are ignored during election. Voters reserve a bond as they vote.
+//! The entire free balance of a voter is locked upon voting and can only be used to pay for fees.
+//! Voters can update their votes by calling `vote()` again during the same round. This does not
+//! effect the reserved bond or lock. After a term, voters _must_ call `remove_voter` to get their
+//! bond back and remove the lock. Otherwise the bond amount will stay reserved and the lock will
+//! persist. Furthermore, voters of an old round cannot vote for the current round until they remove
+//! their previous data via `remove_voter`.
+//!
+//! Candidates also reserve a bond as they submit candidacy. A candidate cannot take their candidacy
+//! back. Winner candidates will be moved to the members list and will get their bond back at end of
+//! their term. Loser candidates will immediately get their bond back. Note that unlike phragmen in
+//! staking, candidates do NOT automatically vote for themselves (though they _could_ via a separate
+//! transaction). Furthermore, the amount of tokens (stake) held by the candidate does not matter.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sr_primitives::traits::{Zero, StaticLookup, Bounded, Convert};
+use sr_primitives::weights::SimpleDispatchInfo;
+use srml_support::{StorageValue, StorageMap, EnumerableStorageMap, decl_storage, decl_event, ensure,
+	decl_module, dispatch,
+	traits::{Currency, Get, LockableCurrency, LockIdentifier, ReservableCurrency, WithdrawReason,
+		WithdrawReasons, ChangeMembers
+	}
+};
+use system::{self, ensure_signed, ensure_root};
+
+const MODULE_ID: LockIdentifier = *b"phrelect";
+
+type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
+pub const DEFAULT_CANDIDACY_BOND: u32 = 9;
+pub const DEFAULT_VOTING_BOND: u32 = 2;
+
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+	/// The currency that people are electing with.
+	type Currency:
+		LockableCurrency<Self::AccountId, Moment=Self::BlockNumber> +
+		ReservableCurrency<Self::AccountId>;
+
+	/// What to do when the members change.
+	type ChangeMembers: ChangeMembers<Self::AccountId>;
+
+	/// Convert a balance into a number used for election calculation.
+	/// This must fit into a `u64` but is allowed to be sensibly lossy.
+	type CurrencyToVote: Convert<BalanceOf<Self>, u64> + Convert<u128, BalanceOf<Self>>;
+
+	/// How much should be locked up in order to submit one's candidacy.
+	type CandidacyBond: Get<BalanceOf<Self>>;
+
+	/// How much should be locked up in order to be able to submit votes.
+	type VotingBond: Get<BalanceOf<Self>>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as PhragmenElection {
+		// ---- parameters
+		/// Number of seats to elect.
```

Comment:
if this changes, does this take effect immediately? or does it only take effect at the next election? if so is there a new election immediately?

Reply:
nope; the change will happen on next elect. Will add a doc for it.